src/liblsquic/lsquic_logger.c

```diff
@@ -143,6 +146,27 @@ print_timestamp (void)
     else if (g_llts == LLTS_CHROMELIKE)
         lsquic_printf("%02d%02d/%02d%02d%02d.%06d ", tm.tm_mon + 1,
             tm.tm_mday,tm.tm_hour, tm.tm_min, tm.tm_sec, (int) tv.tv_usec);
+#else
+    SYSTEMTIME tm = { 0 };
+    GetSystemTime(&tm);
+    if (g_llts == LLTS_YYYYMMDD_HHMMSSUS)
+        lsquic_printf("%04d-%02d-%02d %02d:%02d:%02d.%06d ",
+            tm.wYear + 1900, tm.wMonth + 1, tm.wDay,
+            tm.wHour, tm.wMinute, tm.wSecond, tm.wMilliseconds*1000 );
+    else if (g_llts == LLTS_YYYYMMDD_HHMMSSMS)
+        lsquic_printf("%04d-%02d-%02d %02d:%02d:%02d.%03d ",
+            tm.wYear + 1900, tm.wMonth + 1, tm.wDay,
+            tm.wHour, tm.wMinute, tm.wSecond, tm.wMilliseconds*1000 );
+    else if (g_llts == LLTS_HHMMSSMS)
+        lsquic_printf("%02d:%02d:%02d.%03d ", tm.wHour, tm.wMinute,
+                                    tm.wSecond, tm.wMilliseconds );
+    else if (g_llts == LLTS_HHMMSSUS)
+        lsquic_printf("%02d:%02d:%02d.%03d ", tm.wHour, tm.wMinute,
+                                    tm.wSecond, tm.wMilliseconds*1000 );
+    else if (g_llts == LLTS_CHROMELIKE)
+        lsquic_printf("%02d%02d/%02d%02d%02d.%06d ", tm.wMonth + 1,
+            tm.wDay,tm.wHour, tm.wMinute, tm.wSecond, (int) tm.wMilliseconds*1000);
+#endif
```

Comment:
If we're going to live with millisecond granularity, we could use the portable ftime function.  But gettimeofday gets you microsecond granularity.  So let's implement gettimeofday for Windows (if we're going to add some code, let's add the whole deal).  See: https://social.msdn.microsoft.com/Forums/vstudio/en-US/430449b3-f6dd-4e18-84de-eebd26a8d668/gettimeofday?forum=vcgeneral